library/modules/Units.cpp

```diff
@@ -626,7 +626,8 @@ bool Units::isDwarf(df::unit *unit)
 bool Units::isAnimal(df::unit* unit)
 {
     CHECK_NULL_POINTER(unit)
-    return unit->enemy.caste_flags.is_set(df::enums::caste_raw_flags::NATURAL_ANIMAL);
+    return unit->animal.leave_countdown != 0 || 
+           unit->enemy.caste_flags.is_set(df::enums::caste_raw_flags::NATURAL_ANIMAL);
```

Comment:
I am not convinced that `NATURAL_ANIMAL` is an appropriate check condition. Per the DF wiki, `NATURAL_ANIMAL` is an alias of `NATURAL` and `NATURAL` means:

> Animal is considered to be natural. NATURAL animals will not engage creatures tagged with `AT_PEACE_WITH_WILDLIFE` in combat unless they are members of a hostile entity and vice-versa.

This doesn't seem relevant to the purpose of this predicate.